scripts/bert/finetune_squad.py

```diff
@@ -0,0 +1,459 @@
+import json
+import logging
+import os
+import random
+import time
+import warnings
+
+import numpy as np
+import mxnet as mx
+from mxnet import gluon, nd
+
+import gluonnlp as nlp
+from gluonnlp.data import SQuAD
+from bert_for_qa import BertForQALoss, BertForQA
+from qa_dataset import (SQuADTransform, bert_qa_batchify_fn,
+                        preprocess_dataset)
+from qa_evaluate import get_F1_EM, predictions
+
+np.random.seed(6)
+random.seed(6)
+mx.random.seed(6)
+
+log = logging.getLogger('gluonnlp')
+log.setLevel(logging.DEBUG)
+formatter = logging.Formatter(
+    fmt='%(levelname)s:%(name)s:%(asctime)s %(message)s', datefmt='%H:%M:%S')
+
+
+parser = argparse.ArgumentParser(description='BERT QA example.'
+                                 'We fine-tune the BERT model on SQuAD dataset.')
+
+parser.add_argument('--only_predict',
+                    action='store_true',
+                    help='Whether to predict only.')
+
+parser.add_argument('--model_parameters',
+                    type=str,
+                    default=None,
+                    help='Model parameter file')
+
+parser.add_argument('--model',
+                    type=str,
+                    default='bert_12_768_12',
+                    help='BERT model name. options are bert_12_768_12 and bert_24_1024_16.')
+
+parser.add_argument('--dataset_name',
+                    type=str,
+                    default='book_corpus_wiki_en_uncased',
+                    help='BERT dataset name.'
+                    'options are book_corpus_wiki_en_uncased and book_corpus_wiki_en_cased.')
+
+parser.add_argument('--uncased',
+                    action='store_false',
+                    help='if not set, inputs are converted to lower case.')
+
+parser.add_argument('--output_dir',
+                    type=str,
+                    default='./output_dir',
+                    help='The output directory where the model params will be written.'
+                    ' default is ./output_dir')
+
+parser.add_argument('--epochs',
+                    type=int,
+                    default=2,
+                    help='number of epochs, default is 2')
+
+parser.add_argument('--batch_size',
+                    type=int,
+                    default=12,
+                    help='Batch size. Number of examples per gpu in a minibatch. default is 12')
+
+parser.add_argument('--test_batch_size',
+                    type=int,
+                    default=24,
+                    help='Test batch size. default is 24')
+
+parser.add_argument('--optimizer',
+                    type=str,
+                    default='bertadam',
+                    help='optimization algorithm. default is bertadam')
+
+parser.add_argument('--accumulate',
+                    type=int,
+                    default=None,
+                    help='The number of batches for '
+                    'gradients accumulation to simulate large batch size. Default is None')
+
+parser.add_argument('--lr',
```

Comment:
Because the large model in the original paper is not reproduced, only the results of the pre-training in the original repo are reproduced, whether the default value of the parameter is necessary to be consistent with the original paper, or the parameter value of the original repo that can be reproduced is used.